src/wp-includes/class-wp.php

```diff
@@ -367,6 +367,99 @@ public function parse_request( $extra_query_vars = '' ) {
 			}
 		}
 
+		// Prevent invalid date queries.
+
+		// Array containing all min-max checks.
+		$min_max_checks = array();
+
+		// Months per year.
+		$min_max_checks['monthnum'] = array(
+			'min' => 1,
+			'max' => 12,
+		);
+
+		// Weeks per year.
+		if ( isset( $this->query_vars['year'] ) ) {
+			/*
+			 * If we have a specific year, use it to calculate number of weeks.
+			 * Note: the number of weeks in a year is the date in which Dec 28 appears.
+			 */
+			$week_count = gmdate( 'W', mktime( 0, 0, 0, 12, 28, absint( $this->query_vars['year'] ) ) );
+
+		} else {
+			// Otherwise set the week-count to a maximum of 53.
+			$week_count = 53;
+		}
+
+		$min_max_checks['w'] = array(
+			'min' => 1,
+			'max' => $week_count,
+		);
+
+		// Days per month.
+		$min_max_checks['day'] = array(
+			'min' => 1,
+			'max' => 31,
+		);
+
+		// Hours per day.
+		$min_max_checks['hour'] = array(
+			'min' => 0,
+			'max' => 23,
+		);
+
+		// Minutes per hour.
+		$min_max_checks['minute'] = array(
+			'min' => 0,
+			'max' => 59,
+		);
+
+		// Seconds per minute.
+		$min_max_checks['second'] = array(
+			'min' => 0,
+			'max' => 59,
+		);
+
+		// Concatenate and unset each invalid value.
+
+		foreach ( $min_max_checks as $key => $check ) {
+			if (
+				! array_key_exists( $key, $this->query_vars )
+			) {
+				continue;
+			}
+
+			$is_between = $this->query_vars[ $key ] >= $check['min'] && $this->query_vars[ $key ] <= $check['max'];
+
+			if ( ! is_numeric( $this->query_vars[ $key ] ) || ! $is_between ) {
+				unset( $this->query_vars[ $key ] );
+			}
+		}
+
+		$day_month_year_error_msg = '';
+
+		$day_exists   = array_key_exists( 'day', $this->query_vars ) && is_numeric( $this->query_vars['day'] );
+		$month_exists = array_key_exists( 'monthnum', $this->query_vars ) && is_numeric( $this->query_vars['monthnum'] );
+		$year_exists  = array_key_exists( 'year', $this->query_vars ) && is_numeric( $this->query_vars['year'] );
```

Comment:
**thought (if-minor):** I would change the redundat variables into function also recommend using `isset` it also checks value is not null

```suggestion
            $is_value_exists = function(string $value) {
                return isset( $this->query_vars[$value] ) && is_numeric( $this->query_vars[$value] );
            };
```